src/AutomatedCar/Models/World.cs

```diff
@@ -20,13 +30,184 @@ public AutomatedCar ControlledCar
 
         public int Height { get; set; }
 
+        public List<WorldObject> NPCs { get; set; }
+
+        private World()
+        {
+
+        }
+
         public void AddObject(WorldObject worldObject)
         {
             this.WorldObjects.Add(worldObject);
         }
+
+        private static List<WorldObject> ReadWorldObjects(string filename)
+        {
+            List<WorldObject> allObjects;
+
+            JObject worldObjectsInFile = JObject.Parse(File.ReadAllText("..\\..\\..\\Assets\\" + filename));
+
+            IList<JToken> results = worldObjectsInFile["objects"].Children().ToList();
+
+            allObjects = results.Select(r => new WorldObject
+            {
+                X = (int)r["x"],
+                Y = (int)r["y"],
+                FileName = (string)r["type"],
+                Angle = Math.Acos((double)r["m11"]),
+            }).ToList();
+
+            return allObjects;
+        }
+
+        private static List<WorldObject> ReadPolygons(string filename)
+        {
+            List<WorldObject> polygons;
+
+            JObject polygonsInFile = JObject.Parse(File.ReadAllText("..\\..\\..\\Assets\\" + filename));
+
+            IList<JToken> polygonsInJson = polygonsInFile["objects"].Children().ToList();
+
+            polygons = polygonsInJson.Select(polygon => new WorldObject
+            {
+                FileName = (string)polygon["typename"],
+                Polygon = polygon["polys"].Children()["points"].Select(pointlist => new Polygon
+                {
+                    Points = pointlist.Select(point => new Point((double)point[0], (double)point[1])).ToList(),
+                }).ToList().ToArray(),
+            }).ToList();
+
+            return polygons;
+        }
+
+        private static List<WorldObject> ReadRotationPoints(string filename)
+        {
+            List<WorldObject> rotationPoints;
+
+            var rotationPointsInFile = JArray.Parse(File.ReadAllText("..\\..\\..\\Assets\\" + filename));
+
+            rotationPoints = rotationPointsInFile.Children().ToList().Select(rotationpoint => new WorldObject
+            {
+                FileName = rotationpoint["name"].ToString().Substring(0, rotationpoint["name"].ToString().LastIndexOf(".")),
+                RotationCenterPointX = (int)rotationpoint["x"],
+                RotationCenterPointY = (int)rotationpoint["y"],
+            }).ToList();
+
+            return rotationPoints;
+        }
+
+        public static World GetInstance()
+        {
+            JObject configFilenames = JObject.Parse(File.ReadAllText("..\\..\\..\\Assets\\config.json"));
+
+            var allObjects = ReadWorldObjects(configFilenames["world_objects"].ToString());
+            var onlyPolygons = ReadPolygons(configFilenames["polygons"].ToString());
+            var onlyRotationPoints = ReadRotationPoints(configFilenames["rotation_points"].ToString());
+
+            // got bored with LINQ, could be refactored in necessary
```

Comment:
Remek... Ez a komment több szempontból is szükségtelen. Kb. azt mondja, hogy tudom, hogy lehetne jobban, de ha nem tetszik akkor csináld meg magad.

Ha tudod, hogy nem jó, nem elég jó, akkor legalább vegyél fel taszkot arra, hogy valaki csinálja meg.

Reply:
Rendben, kiszedtem, többet nem fordul elő.

Reply:
Hétfőn beszéltük, hogy ez egy work-in-progress PR volt,ilyenkor célszerű draft-nak jelölni és akkor látom, hogy nem véglegesnek van szánva a dolog.